File: c#/tema_5/ConcertPro/Service/OrderService.cs
using ConcertPro.Dao.Exception;
using ConcertPro.Dao.Model;
using ConcertPro.Dao.Repository.OrderRepository;
using ConcertPro.Dao.Repository.SpectacleRepository;
using log4net;

namespace ConcertPro.Service
{
    public class OrderService
    {
        private readonly IOrderRepository _orderRepository;
        private readonly ISpectacleRepository _spectacleRepository;
        private static readonly ILog Log = LogManager.GetLogger("OrderService");

        public OrderService(IOrderRepository orderRepository, ISpectacleRepository spectacleRepository)
        {
            Log.InfoFormat("Initializing OrderService");
            
            _orderRepository = orderRepository;
            _spectacleRepository = spectacleRepository;
        }

        public void AddOrder(Order order)
        {
            Log.InfoFormat("Initializing the AddOrder task.");
            
            _orderRepository.Save(order);

            Spectacle spectacle = order.Spectacle;

            if (order.Spectacle.SeatsAvailable < order.NumberOfSeats)
            {
                throw new ValidationException("There are not enough seats for this spectacle!");
            }

            spectacle.SeatsSold += order.NumberOfSeats;
            spectacle.SeatsAvailable -= order.NumberOfSeats;
            _spectacleRepository.Update(spectacle, spectacle.Id);
        }
    }
}

File: java/tema_5/src/main/java/ro/mpp2024/concertpro/service/OrderService.java
package ro.mpp2024.concertpro.service;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.mpp2024.concertpro.dao.exception.ValidationException;
import ro.mpp2024.concertpro.dao.model.Order;
import ro.mpp2024.concertpro.dao.model.Spectacle;
import ro.mpp2024.concertpro.dao.repository.order_repository.IOrderRepository;
import ro.mpp2024.concertpro.dao.repository.spectacle_repository.ISpectacleRepository;

public class OrderService {
    private final IOrderRepository orderRepository;
    private final ISpectacleRepository spectacleRepository;
    private static final Logger logger = LogManager.getLogger();

    public OrderService(IOrderRepository orderRepository, ISpectacleRepository spectacleRepository) {
        logger.info("Initializing Order Service.");
        this.orderRepository = orderRepository;
        this.spectacleRepository = spectacleRepository;
    }

    public void addOrder(Order order) {
        this.orderRepository.save(order);

        Spectacle newSpectacle = order.getSpectacle();

        if(order.getSpectacle().getSeatsAvailable() < order.getNumberOfSeats()) {
            throw new ValidationException("There are not enough seats for this spectacle.");
        }

        newSpectacle.setSeatsSold(newSpectacle.getSeatsSold() + order.getNumberOfSeats());
        newSpectacle.setSeatsAvailable(newSpectacle.getSeatsAvailable() - order.getNumberOfSeats());
        this.spectacleRepository.update(newSpectacle, newSpectacle.getId());
    }
}
